feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch

```diff
@@ -0,0 +1,1489 @@
+===================================================================
+--- opensync-2.0.5.0.orig/vendor/tip/src/lib/target/inc/target_tip.h
++++ opensync-2.0.5.0/vendor/tip/src/lib/target/inc/target_tip.h
+@@ -29,6 +29,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
+ 
+ #include "dpp_client.h"
+ #include "dpp_survey.h"
++#include "dpp_events.h"
+ 
+ #define TARGET_CERT_PATH            "/var/run/openvswitch/certs"
+ #define TARGET_MANAGERS_PID_PATH    "/tmp/dmpid"
+@@ -54,6 +55,12 @@ typedef struct
+ 	uint32_t duration_ms;
+ } target_survey_record_t;
+ 
++typedef struct
++{
++        dpp_event_record_t record;
++
++} target_event_record_t;
++
+ typedef void target_capacity_data_t;
+ 
+ /******************************************************************************
+Index: opensync-2.0.5.0/src/lib/target/inc/target_common.h
+===================================================================
+--- opensync-2.0.5.0.orig/src/lib/target/inc/target_common.h
++++ opensync-2.0.5.0/src/lib/target/inc/target_common.h
+@@ -389,6 +389,13 @@ typedef bool target_stats_clients_cb_t (
+         void                       *ctx,
+         int                         status);
+ 
++/******************************************************************************
++ *  EVENT definitions
++ *****************************************************************************/
++target_event_record_t *target_event_record_alloc();
++bool target_stats_events_get(radio_entry_t *radio_cfg,
++                             ds_dlist_t *record_list);
++
+ /**
+  * @brief Get clients stats
+  *
+Index: opensync-2.0.5.0/src/sm/src/sm.h
+===================================================================
+--- opensync-2.0.5.0.orig/src/sm/src/sm.h
++++ opensync-2.0.5.0/src/sm/src/sm.h
+@@ -161,6 +161,13 @@ bool sm_network_probe_report_request(
+         sm_stats_request_t         *request);
+ 
+ /******************************************************************************
++ *  EVENTS REPORT definitions
++ *****************************************************************************/
++bool sm_events_report_request(
++        radio_entry_t              *radio_cfg,
++        sm_stats_request_t         *request);
++
++/******************************************************************************
+  *  SURVEY_REPORT definitions
+  *****************************************************************************/
+ bool sm_survey_report_request(
+@@ -271,6 +278,7 @@ typedef enum
+     STS_REPORT_DEVICE,
+     STS_REPORT_RSSI,
+     STS_REPORT_NETWORK_PROBE,
++    STS_REPORT_EVENT,
+     STS_REPORT_MAX,
+     STS_REPORT_ERROR = STS_REPORT_MAX
+ } sm_report_type_t;
+Index: opensync-2.0.5.0/src/sm/src/sm_events_report.c
+===================================================================
+--- /dev/null
++++ opensync-2.0.5.0/src/sm/src/sm_events_report.c
+@@ -0,0 +1,238 @@
++/*
++Copyright (c) 2015, Plume Design Inc. All rights reserved.
```

Comment:
is this file from Plume ? if not add bsd 3clause SPDX instead

Reply:
No, this is one of mine. It was a copy/paste from the one from Plume, hence the wrong licence.